apollo-runtime/src/main/java/com/apollographql/android/cache/normalized/GraphQLResponseNormalizer.java

```diff
@@ -0,0 +1,137 @@
+package com.apollographql.android.cache.normalized;
+
+import com.apollographql.android.api.graphql.Field;
+import com.apollographql.android.api.graphql.Operation;
+import com.apollographql.android.cache.ResponseReaderShadow;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.Stack;
+
+import javax.annotation.Nullable;
+
+public class GraphQLResponseNormalizer implements ResponseReaderShadow {
+
+  private Stack<List<String>> pathStack;
+  private Stack<ApolloRecord> recordStack;
+  private Stack<Object> valueStack;
+  private Set<String> dependentKeys;
+  private final CacheKeyResolver cacheKeyResolver;
+
+  private List<String> path;
+  private ApolloRecord currentRecord;
+
+  private InMemoryNormalizedCacheStore recordSet;
+
+  public Collection<ApolloRecord> getRecords() {
+    return recordSet.getAllRecords();
+  }
+
+  public Set<String> getDependentKeys() {
+    return dependentKeys;
+  }
+
+  GraphQLResponseNormalizer(CacheKeyResolver cacheKeyResolver) {
+    this.cacheKeyResolver = cacheKeyResolver;
+  }
+
+  private static String rootKeyForOperation(Operation operation) {
+    //Todo: differentiate queries and mutations
+    return "QUERY_ROOT";
+  }
+
+  @Override public void willResolveRootQuery(Operation operation) {
+    pathStack = new Stack<>();
+    recordStack = new Stack<>();
+    valueStack = new Stack<>();
+    dependentKeys = new HashSet<>();
+
+    path = new ArrayList<>();
+    currentRecord = new ApolloRecord(rootKeyForOperation(operation));
+    recordSet = new InMemoryNormalizedCacheStore();
```

Comment:
Is this intentionally, use `InMemoryNormalizedCacheStore `  shouldn't it be configured in constructor what impl of `NormalizedCacheStore` will be used?